File: nim/rna-transcription/rna_transcription.nim

proc toRna(c: char): char =
  case c:
    of 'G': 'C'
    of 'C': 'G'
    of 'T': 'A'
    of 'A': 'U'
    else: raise newException(ValueError, "invalid nucleotide")

proc toRna*(str: string): string =
  for c in str:
    result.add(c.toRna())

File: typescript/rna-transcription/rna-transcription.ts
export function toRna(nucleotides: string): string {

  let result = '';
  for (let nucleotide of nucleotides) {
    if (!AllowedDna.includes(nucleotide))
      throw new Error('Invalid input DNA.');
    
    result += RnaNucleotides[DnaNucleotides.indexOf(nucleotide)];
  }
  
  return result;
}

const DnaNucleotides = ['G', 'C', 'T', 'A']
const RnaNucleotides = ['C', 'G', 'A', 'U']

const AllowedDna = DnaNucleotides.join('')
